apps/hash-frontend/src/pages/types/[[...type-kind]].page.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+  DataTypeRootType,
+  DataTypeWithMetadata,
+  PropertyTypeRootType,
+  PropertyTypeWithMetadata,
+} from "@local/hash-subgraph";
+import { getRoots } from "@local/hash-subgraph/stdlib";
+import { Box, Container, Typography } from "@mui/material";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { NextSeo } from "next-seo";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { useBlockProtocolQueryDataTypes } from "../../components/hooks/block-protocol-functions/ontology/use-block-protocol-query-data-types";
+import { useBlockProtocolQueryPropertyTypes } from "../../components/hooks/block-protocol-functions/ontology/use-block-protocol-query-property-types";
+import { useLatestEntityTypesOptional } from "../../shared/entity-types-context/hooks";
+import { isLinkEntityType } from "../../shared/entity-types-context/util";
+import { FilesLightIcon } from "../../shared/icons/files-light-icon";
+import { getLayoutWithSidebar, NextPageWithLayout } from "../../shared/layout";
+import { TopContextBar } from "../shared/top-context-bar";
+import { tabTitles, TypesPageTabs } from "./types-page-tabs";
+import { TypesTable } from "./types-table";
+
+const parsedQueryParams = [
+  "entity-type",
+  "link-type",
+  "property-type",
+  "data-type",
+] as const;
+
+type ParsedQueryKindParam = (typeof parsedQueryParams)[number];
+
+type ParsedQueryParams = {
+  ["type-kind"]?: ParsedQueryKindParam[];
+};
+
+export const getStaticPaths: GetStaticPaths<ParsedQueryParams> = () => ({
+  paths: [
+    { params: { "type-kind": [] } },
+    ...parsedQueryParams.map((kind) => ({ params: { "type-kind": [kind] } })),
+  ],
+  fallback: false,
+});
+
+export type TabId = "all" | ParsedQueryKindParam;
+
+type TypesPageProps = {
+  currentTab: TabId;
+};
+
+export const getStaticProps: GetStaticProps<
+  TypesPageProps,
+  ParsedQueryParams
+> = ({ params }) => {
+  const currentTab = params?.["type-kind"]?.[0] ?? "all";
+
+  return { props: { currentTab } };
+};
+
+const contentMaxWidth = 1000;
+
+const TypesPage: NextPageWithLayout<TypesPageProps> = ({ currentTab }) => {
+  const latestEntityTypes = useLatestEntityTypesOptional({
+    includeArchived: true,
+  });
+
+  const latestNonLinkEntityTypes = useMemo(
+    () =>
+      latestEntityTypes?.filter(
+        (entityType) => !isLinkEntityType(entityType.schema),
+      ),
+    [latestEntityTypes],
+  );
+
+  const latestLinkEntityTypes = useMemo(
+    () =>
+      latestEntityTypes?.filter((entityType) =>
+        isLinkEntityType(entityType.schema),
+      ),
+    [latestEntityTypes],
+  );
+
+  const { queryPropertyTypes } = useBlockProtocolQueryPropertyTypes();
```

Comment:
There's a `useLatestPropertyTypesContextValue` hook you could possibly replace this with, as it does a similar thing.

It might be even better to move the `<LatestPropertyTypesContextProvider>` up from the type page to surround the app, so that we're sharing the same set of property types rather than refetching them on different pages

Reply:
Hmm yes I can move this up so I can make use of the context. Do you know why we return the property types in an object rather than an array by chance?

Reply:
The entity editor asks for them that way, I believe it was done like this to make it easier to find relevant property types easily

Reply:
Alright I suppose it doesn't make much difference, may have been for a slight performance gain when looking up property types by versioned URL. Have made it a global context, and re-used the value in https://github.com/hashintel/hash/pull/2908/commits/e638dee00ac1ae9e00db280fc204740fd6d039b2

Reply:
Cool, looks good